### PR TITLE
View web account password

### LINKED ISSUE #7: View Password based on Web Account
Using the Terminal, a user should be able to view a password based on an account that they request so that they may use it to log in to that account.

### PR DESCRIPTION
**Pull request by ryanloutos-wustl**

### CODE REVIEW

File: src/websiteAccount/WebsiteAccount.java [hunk 1]
```diff
@@ -124,46 +132,68 @@ public static boolean addWebsiteAccountPrompts(ConnectionSource databaseConnecti
...
-					}
-				}
-				// if the account they typed in doesn't exist, login will never be updated and continue
-				//	to be an empty string, so this writes a little prompt telling the user to enter a 
-				//	re enter the account
-				if (login == "") {
-					System.out.println("Invalid account name. Type the name exactly how it printed above.");
+				if (nickname.equals(Encryption.decrypt(account.getNickname()))) {
+					accountExists = true;
+					returnAccountInfo = account.toString();
```
Comment (taliaweiss1) [inline src/websiteAccount/WebsiteAccount.java:192]:
```suggestion
					returnAccountInfo = account.toString();
					break;
```

File: src/websiteAccount/WebsiteAccount.java [hunk 2]
```diff
@@ -124,46 +132,68 @@ public static boolean addWebsiteAccountPrompts(ConnectionSource databaseConnecti
...
+	 * @return - what was printed for that specific website account...mainly used for testing
+	 */
+	public static String viewWebsiteAccountInfo(ConnectionSource databaseConnection, Scanner keyboard, User safeStoreUser) {
+		printAllWebsiteAccounts(databaseConnection, safeStoreUser);
+		ForeignCollection<WebsiteAccountEntity> websiteAccounts = getAllWebsiteAccounts(databaseConnection, safeStoreUser);
+
+		String returnAccountInfo = "No accounts";
+		Boolean accountExists = false;
+		System.out.println("Type the account that you want to see the login info for:");
+		while (!accountExists) {
```
Comment (taliaweiss1) [inline src/websiteAccount/WebsiteAccount.java:187]: Should there be a way for this while loop to terminate if the user doesn't have any accounts saved (in which case the call to printAllWebsiteAccounts(databaseConnection, safeStoreUser would print No accounts but then there's no way for the user to get out of this while loop)

File: test/websiteAccountTests/WebsiteAccountTests.java [hunk 3]
```diff
@@ -128,19 +126,56 @@ void testInsertWebsiteAccountToDatabaseWithSameNickname() {
...
-		File file = new File("test/queryWithDatabaseTests/viewLoginForWebAccount.txt");
+	void testGetAllWebsiteAccountsForUser() {
 		try {
 			// add user and account to db
 			User testUser = new User(testUserUsername, testUserPassword);
-			testUser.createSafeStoreAccount(databaseConnection);
+			testUser.createSafeStoreAccountThroughDatabase(databaseConnection);
+			WebsiteAccount account = new WebsiteAccount(testUser, testAccountNickname, testAccountLogin, testAccountPassword);
+			account.addWebsiteAccount(databaseConnection);
+			WebsiteAccount account2 = new WebsiteAccount(testUser, "anothaOne", "thisIsALogin", "screw_camel_case");
```
Comment (taliaweiss1) [inline test/websiteAccountTests/WebsiteAccountTests.java:136]: lol

**Review by taliaweiss1 — COMMENTED** [review]
Looks great! Refer to comments in edited files

File: test/websiteAccountTests/WebsiteAccountTests.java (hunk 3 above)
Comment (ryanloutos-wustl) [reply test/websiteAccountTests/WebsiteAccountTests.java:136]: ;)

File: src/websiteAccount/WebsiteAccount.java (hunk 2 above)
Comment (ryanloutos-wustl) [reply src/websiteAccount/WebsiteAccount.java:187]: Yes you are right....I added an if statement that if there are no accounts, the method will return early

File: src/websiteAccount/WebsiteAccount.java (hunk 1 above)
Comment (ryanloutos-wustl) [reply src/websiteAccount/WebsiteAccount.java:192]: Fixed

**Review by taliaweiss1 — APPROVED** [review]
looks great!

File: src/tables/WebsiteAccountEntity.java [hunk 4]
```diff
@@ -4,6 +4,8 @@
 import com.j256.ormlite.misc.BaseDaoEnabled;
```
Comment (mbendy) [inline src/tables/WebsiteAccountEntity.java:4]: I can't leave a comment on app.db, but just want to be sure that nothing is diff besides you messing around w rows?

File: src/websiteAccount/WebsiteAccount.java [hunk 5]
```diff
@@ -124,46 +132,73 @@ public static boolean addWebsiteAccountPrompts(ConnectionSource databaseConnecti
...
+	/**
+	 * Prompts the user to pick the website account they want to view the login and password info for and prints it out
+	 * @param databaseConnection
+	 * @param keyboard
+	 * @param safeStoreUser
+	 * @return - what was printed for that specific website account...mainly used for testing
+	 */
+	public static String viewWebsiteAccountInfo(ConnectionSource databaseConnection, Scanner keyboard, User safeStoreUser) {
+		printAllWebsiteAccounts(databaseConnection, safeStoreUser);
+		ForeignCollection<WebsiteAccountEntity> websiteAccounts = getAllWebsiteAccounts(databaseConnection, safeStoreUser);
```
Comment (mbendy) [inline src/websiteAccount/WebsiteAccount.java:178]: i dont know if its too late, but like why do you get all website accounts? Seems like a lot of work each time we want to view information on a website account? Shouldn't we just query for the specific one they ask for?

File: src/websiteAccount/WebsiteAccount.java [hunk 6]
```diff
@@ -124,46 +132,73 @@ public static boolean addWebsiteAccountPrompts(ConnectionSource databaseConnecti
...
+	
+	/**
+	 * Prompts the user to pick the website account they want to view the login and password info for and prints it out
+	 * @param databaseConnection
+	 * @param keyboard
+	 * @param safeStoreUser
+	 * @return - what was printed for that specific website account...mainly used for testing
+	 */
+	public static String viewWebsiteAccountInfo(ConnectionSource databaseConnection, Scanner keyboard, User safeStoreUser) {
+		printAllWebsiteAccounts(databaseConnection, safeStoreUser);
```
Comment (mbendy) [inline src/websiteAccount/WebsiteAccount.java:177]: This isn't in the scope of this ticket, please delete this for consistency among credit cards/debit cards/website accounts (this functionality will be added at a later iteration)

File: src/websiteAccount/WebsiteAccount.java [hunk 7]
```diff
@@ -124,46 +132,73 @@ public static boolean addWebsiteAccountPrompts(ConnectionSource databaseConnecti
...
+					returnAccountInfo = account.toString();
+					break;
 				}
 			}
-			return login;
-		} catch (SQLException e) {
-			e.printStackTrace();
-			return "SQL Error Occured";
+			if (!accountExists) {
+				System.out.println("Invalid account name. Type the name exactly how it is printed above.");
```
Comment (mbendy) [inline src/websiteAccount/WebsiteAccount.java:197]: since we don't want to print all accounts just ask to retype

**Review by mbendy — CHANGES_REQUESTED** [review]

**Review by mbendy — APPROVED** [review]

**Pull request merged**